### DISCUSSION TITLE
Query the table data. If there is a foreign key table, an error Unknown column will be reported

### DISCUSSION THREAD

**Discussion by Shen120** [sequelize/sequelize#15154] [Help & Questions]
## Bug Description

1. There are two tables: user and card. The cardTypeID of the user is associated with the ID field of the card, forming a foreign key
2. An error occurred while using the query; Unknown column 'Card Id 'in' field list ', attached figure:
3. 
![image](https://user-images.githubusercontent.com/33613409/193740311-eaab14a4-7dc8-4d6e-9789-9ce3e96e56bb.png)
4. After troubleshooting: there are related fields in the database, and the spelling is correct. The SQL statement generated during query is correct

### Reproducible Example

[Example](https://github.com/Shen120/egg-prod-bug)

### What is actually happening?

Normal query without query error

### Environment

- Sequelize version: v6
- Node.js version: v.14.17.5
- If TypeScript related: TypeScript version: 4.8.3
- Database & Version: MySql 5.7.28

**Comment by Shen120** [comment]
![image](https://user-images.githubusercontent.com/33613409/193744237-dea27256-b023-4290-94d4-f674ef1a6b0e.png)

**Comment by Shen120** [comment]
![image](https://user-images.githubusercontent.com/33613409/193744472-a58ce4fd-e3b1-4006-8aad-b0c09007c46c.png)

**Comment by Shen120** [comment]
Find the reason
```
await ctx.model.NonMemberConsumption.findAndCountAll({
      ...ctx.helper.computePage(params.page, params.pageSize),
      where: {
        ...ctx.helper.transOpParams(params, {
          orderNo: {condition: "eq"},
          goodsID: {condition: "eq"},
          createTime: {condition: "between"},
        })
      },
      include: [
        {
          model: app.model.Commodity,
          attributes: [],
          required: true
        }
      ],
      attributes: {
        exclude: ["goodsID"],
        include: [
          // TODO The reason for the query error after deployment is again, and the specific scheme is to be determined
          [col("Commodity.id"), "goodsId"],
          [col("Commodity.name"), "goodsName"],
          [col("Commodity.code"), "goodsCode"],
        ]
      },
      distinct: true,
      order: [
        ["createTime", "DESC"]
      ]
    })
```
There is no specific solution to the error caused by using the include attribute of attributes

**Comment by Shen120** [comment]
Can someone help me

**Comment by ephys** [comment]
Hello,

You should aim to provide a reproducible example that is as minimal as possible. Not doing so makes it much less likely to receive help

Anyway your issue is definitely that you're selecting rows of an association inside of your top level `attributes` option. The attributes of your include must go in the `attributes` of option of that include

**Reply by Shen120** [reply]
> Hello,
> 
> You should aim to provide a reproducible example that is as minimal as possible. Not doing so makes it much less likely to receive help
> 
> Anyway your issue is definitely that you're selecting rows of an association inside of your top level `attributes` option. The attributes of your include must go in the `attributes` of option of that include

I provided a project here
![image](https://user-images.githubusercontent.com/33613409/196906744-d0b9a50f-d181-4db0-a816-279c71e8d22f.png)

**Reply by ephys** [reply]
The important part being "as minimal as possible", as the people providing help do so voluntarily and are much more likely to help if the example doesn't include unrelated things

**Reply by Shen120** [reply]
> The important part being "as minimal as possible", as the people providing help do so voluntarily and are much more likely to help if the example doesn't include unrelated things

Oh, god! The contents of git warehouse are wrong，and I have updated them；sorry

**Reply by Shen120** [reply]
This sql statement is incorrect
```
SELECT `user`.`id`, `user`.`nickName`, `user`.`avatar`, `user`.`tel`, `user`.`createTime`, `user`.`balance`, `user`.`IDCard`, `user`.`cardTypeID`, `user`.`bindWX`, `user`.`loginTime`, `user`.`lastRechargeTime`, `Card`.`id` AS `cardId`, `Card`.`code` AS `cardCode`, `Card`.`name` AS `cardName` FROM `user` AS `user` INNER JOIN `card` AS `card` ON `user`.`cardTypeID` = `card`.`id` ORDER BY `user`.`createTime` DESC;
```

**Reply by Shen120** [reply]
This error will occur only after deployment. Local development is normal
